users/urls.py

```diff
@@ -1,29 +1,39 @@
 from django.urls import path, include
-from rest_framework_simplejwt.views import TokenBlacklistView
-from .views import CustomTokenObtainPairView
-from .views import CustomPasswordResetView, CustomPasswordResetConfirmView
+from rest_framework_simplejwt.views import TokenRefreshView, TokenBlacklistView
+from .views import (
+    CustomTokenObtainPairView,
+    UserRegistrationView,
+    VerifyEmailView,
+    OAuthTokenObtainPairView,
+    CustomPasswordResetView,
+    CustomPasswordResetConfirmView,
+    ResendEmailView,
+)
 
 
 urlpatterns = [
-    # ----------------------------------------
-    # Custom password recovery endpoints
-    # ----------------------------------------
-    path('reset_password/', CustomPasswordResetView.as_view(), name="custom_reset_password"),
-    path('reset_password_confirm/', CustomPasswordResetConfirmView.as_view(), name="custom_reset_password_confirm"),
-
-    # ----------------------------------------
-    # Custom authentication endpoint
-    # ----------------------------------------
-    path('login/', CustomTokenObtainPairView.as_view(), name='custom_login'),
-
-    # ----------------------------------------
-    # Djoser authentication endpoints
-    # ----------------------------------------
-    path('auth/', include('djoser.urls')),
-    path('auth/', include('djoser.urls.jwt')),
-
-    # ----------------------------------------
-    # JWT Logout endpoint
-    # ----------------------------------------
-    path('auth/jwt/logout/', TokenBlacklistView.as_view(), name='token_blacklist'),
+    # Sign up
+    path('register/', UserRegistrationView.as_view(), name='user_register'),
+
+    # JWT Auth
+    path('jwt/create/', CustomTokenObtainPairView.as_view(), name='jwt-create'),
+    path('jwt/refresh/', TokenRefreshView.as_view(), name='jwt-refresh'),
+    path('jwt/logout/', TokenBlacklistView.as_view(), name='token_blacklist'),
+
+    # Password reset
+    path('password/reset/', CustomPasswordResetView.as_view(), name='custom_reset_password'),
+    path('password/reset/confirm/', CustomPasswordResetConfirmView.as_view(), name='custom_reset_password_confirm'),
+
+    # Email verification
+    path('verify-email/<int:user_id>/<str:token>/', VerifyEmailView.as_view(), name='verify-email'),
+
+    # Resend verification email
+    path('resend-email/', ResendEmailView.as_view(), name='resend-email'),
+
+    # OAuth
+    path('oauth/login/', OAuthTokenObtainPairView.as_view(), name='oauth_login'),
+
+    # Djoser (optional)
+    # path('djoser/', include('djoser.urls')),
+    # path('djoser/', include('djoser.urls.jwt')),
```

Comment:
If Djoser endpoints are commented out because they’re deprecated or replaced, consider removing them or adding a comment explaining why they’re excluded.